### ISSUE TITLE
Entry Editor source - Deleting files and switching to required fields results in Exception

### ISSUE THREAD

**Issue by Siedlerchr** [JabRef/jabref#3369]
current master

Have a biblatex entry with many linked files.
Go to source tab:  Remove file key and its values completely
Go to required Tab: Exception  occurs:

@lenhard  You reworked that  entry editor source tab with the position, right?

```
22:05:40.283 [JavaFX Application Thread] ERROR org.jabref.FallbackExceptionHandler - Uncaught exception occurred in Thread[JavaFX Application Thread,5,main]
java.lang.IndexOutOfBoundsException: 950 not in [0, 1]
	at org.reactfx.util.Lists.checkPosition(Lists.java:110) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.util.Lists.checkPosition(Lists.java:105) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.util.FingerTree$NonEmptyFingerTree.locateProgressively(FingerTree.java:51) ~[reactfx-2.0-M5.jar:?]
	at org.fxmisc.richtext.model.ReadOnlyStyledDocument$Pos.offsetBy(ReadOnlyStyledDocument.java:378) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.richtext.model.ReadOnlyStyledDocument.offsetToPosition(ReadOnlyStyledDocument.java:207) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.richtext.model.GenericEditableStyledDocumentBase.offsetToPosition(GenericEditableStyledDocumentBase.java:119) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.richtext.model.SimpleEditableStyledDocument.offsetToPosition(SimpleEditableStyledDocument.java:7) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.richtext.GenericStyledArea.lambda$new$7(GenericStyledArea.java:664) ~[richtextfx-0.7-M5.jar:?]
	at org.reactfx.value.Val$2.computeValue(Val.java:705) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.ValBase.getValue(ValBase.java:17) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.ValBase.newObserver(ValBase.java:40) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.ValBase.newObserver(ValBase.java:8) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:110) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addInvalidationObserver(Val.java:52) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addListener(Val.java:70) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.observeInvalidations(Val.java:432) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.MappedVal.connect(MappedVal.java:28) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.ValBase.observeInputs(ValBase.java:26) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:108) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.observe(ObservableBase.java:100) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.observeInvalidations(Val.java:61) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val$1.observeInputs(Val.java:104) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:108) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.observe(ObservableBase.java:100) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.EventStream.subscribe(EventStream.java:53) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.SuspendableBase.observeInputs(SuspendableBase.java:49) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:108) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addInvalidationObserver(Val.java:52) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addListener(Val.java:70) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val$2.connect(Val.java:693) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.ValBase.observeInputs(ValBase.java:26) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:108) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addInvalidationObserver(Val.java:52) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.value.Val.addListener(Val.java:80) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.EventStreams$3.observeInputs(EventStreams.java:106) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.addObserver(ObservableBase.java:108) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.ObservableBase.observe(ObservableBase.java:100) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.EventStream.subscribe(EventStream.java:53) ~[reactfx-2.0-M5.jar:?]
	at org.fxmisc.richtext.GenericStyledArea.createCell(GenericStyledArea.java:1375) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.richtext.GenericStyledArea.lambda$new$13(GenericStyledArea.java:760) ~[richtextfx-0.7-M5.jar:?]
	at org.fxmisc.flowless.CellPool.getCell(CellPool.java:20) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.CellListManager.cellForItem(CellListManager.java:75) ~[flowless-0.5.2.jar:?]
	at org.reactfx.collection.MappedList.get(MappedList.java:27) ~[reactfx-2.0-M5.jar:?]
	at org.reactfx.collection.MemoizationListImpl.get(MemoizationList.java:99) ~[reactfx-2.0-M5.jar:?]
	at org.fxmisc.flowless.CellListManager.getCell(CellListManager.java:64) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.CellPositioner.getSizedCell(CellPositioner.java:129) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.CellPositioner.placeStartAt(CellPositioner.java:95) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.Navigator.placeStartAtMayCrop(Navigator.java:183) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.Navigator.visit(Navigator.java:111) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.StartOffStart.accept(TargetPosition.java:49) ~[flowless-0.5.2.jar:?]
	at org.fxmisc.flowless.Navigator.layoutChildren(Navigator.java:67) ~[flowless-0.5.2.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1087) ~[jfxrt.jar:?]
	at org.fxmisc.flowless.VirtualFlow.layoutChildren(VirtualFlow.java:165) ~[flowless-0.5.2.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1087) ~[jfxrt.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1093) ~[jfxrt.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1093) ~[jfxrt.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1093) ~[jfxrt.jar:?]
	at javafx.scene.Parent.layout(Parent.java:1093) ~[jfxrt.jar:?]
	at javafx.scene.Scene.doLayoutPass(Scene.java:552) ~[jfxrt.jar:?]
	at javafx.scene.Scene$ScenePulseListener.pulse(Scene.java:2397) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.Toolkit.lambda$runPulse$29(Toolkit.java:398) ~[jfxrt.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_152]
	at com.sun.javafx.tk.Toolkit.runPulse(Toolkit.java:397) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.Toolkit.firePulse(Toolkit.java:424) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:518) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulse(QuantumToolkit.java:498) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.pulseFromQueue(QuantumToolkit.java:491) ~[jfxrt.jar:?]
	at com.sun.javafx.tk.quantum.QuantumToolkit.lambda$runToolkit$403(QuantumToolkit.java:319) ~[jfxrt.jar:?]
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95) ~[jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method) ~[jfxrt.jar:?]
	at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177) ~[jfxrt.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
```

**Comment by lenhard** [comment]
If you refer to #3360, then no. This only removes the faulty update from the source tab when the selection in the main table changes, but does not touch caret positioning.

Nevertheless, the error message reads very much like a caret positioning problem in the `CodeArea`. There seem to be different attempts at fixing the source tab underway, e.g. #3366. Please check if they fix this when you review one of these.

**Comment by tobiasdiez** [comment]
This should be fixed now with #3366 in the latest development version. Could you please check the build from http://builds.jabref.org/master/. Thanks!
